RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc

```diff
@@ -99,6 +123,14 @@ void PatternRecognitionbyCA::makeTracksters(const PatternRecognitionAlgoBase::In
     //if a seeding region does not lead to any trackster
     tmp.seedID = input.regions[0].collectionID;
     tmp.seedIndex = seedIndices[tracksterId];
+
+    std::pair<float, float> timeTrackster(-99., -1.);
+    if (times.size() >= 3) {
```

Comment:
Any specific reason why you need at least 3 measurements? Maybe it would make sense to "hide" this into the `timeEstimator` code? And make the number 3 a `const` there?

Reply:
3 was tuned looking at the full shower performance using recHits (TDR time).

It is kept and propagated in the chain, as reasonable to maximize efficiency and still allow some clustering of the information (truncation and weighted mean)
I don't exclude that this could be tuned in the future and it might also be different for layerClusters and tracksters.

What if I replace this by a configurable parameter, leaving it in the calling code?

Reply:
Ciao @amartelli maybe we could do both? If you think the parameter could be different for different use cases, then it makes sense to leave it to the caller and make it parametrizable. Yet, maybe you could include it as a parameter into the `fixSizeHighestDensity` method so that the caller does not need to add additional checks on his side?
You could default it if you do not feel like chasing all places in `CMSSW` where this method is already called (if there are few places, it's better to add it explicitly).

Reply:
I agree, it's reasonable,  I'll change it.  thanks